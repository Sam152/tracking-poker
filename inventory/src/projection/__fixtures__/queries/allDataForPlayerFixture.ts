export const allDataForPlayerFixture = [
    {
        date: "2023/06/15",
        entity_type: "player_appearance",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "appearance#player#sashimi#",
        pk: "player#sashimi",
        player: "sashimi",
        player_name: "SASHIMI",
        show: "dVs7ORBHsF0",
        show_name: "Sashimi, Kiran C2B, Israeli Ron & Mike X Play $10/20/40 - Commentary by Christian Soto",
        sk: "appearance#slug#dVs7ORBHsF0#",
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#cc#player#sashimi#",
        gsi2pk: "stat_type#cc",
        gsi2sk: "stat#player#sashimi#slug#dVs7ORBHsF0#",
        pk: "player#sashimi",
        player: "sashimi",
        player_name: "SASHIMI",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#cc#show#sashimi#",
        type: "cc",
        value: 34840,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#cw#player#sashimi#",
        gsi2pk: "stat_type#cw",
        gsi2sk: "stat#player#sashimi#slug#dVs7ORBHsF0#",
        pk: "player#sashimi",
        player: "sashimi",
        player_name: "SASHIMI",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#cw#show#sashimi#",
        type: "cw",
        value: 5840,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#pfr#player#sashimi#",
        gsi2pk: "stat_type#pfr",
        gsi2sk: "stat#player#sashimi#slug#dVs7ORBHsF0#",
        pk: "player#sashimi",
        player: "sashimi",
        player_name: "SASHIMI",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#pfr#show#sashimi#",
        type: "pfr",
        value: 20,
    },
    {
        entity_type: "player_stat",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "stat#stat_type#vpip#player#sashimi#",
        gsi2pk: "stat_type#vpip",
        gsi2sk: "stat#player#sashimi#slug#dVs7ORBHsF0#",
        pk: "player#sashimi",
        player: "sashimi",
        player_name: "SASHIMI",
        show: "dVs7ORBHsF0",
        sk: "stat#stat_type#vpip#show#sashimi#",
        type: "vpip",
        value: 36,
    },
];
