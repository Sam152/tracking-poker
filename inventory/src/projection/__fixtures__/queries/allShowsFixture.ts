export const allShowsFixture = [
    {
        date: "2023/06/15",
        duration: 20165,
        entity_type: "show",
        gsi1pk: "slug#dVs7ORBHsF0",
        gsi1sk: "show#",
        id: "dVs7ORBHsF0",
        operator: "hcl",
        pk: "operator#hcl",
        show_name: "Sashimi, Kiran C2B, Israeli Ron & Mike X Play $10/20/40 - Commentary by Christian Soto",
        sk: "show#date#2023/06/15#slug#dVs7ORBHsF0#",
    },
];
