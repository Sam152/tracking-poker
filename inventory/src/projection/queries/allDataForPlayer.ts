import { QueryCommand } from "@aws-sdk/lib-dynamodb";
import { StructuredQuery } from "./index";
import * as Stat from "../entity/stat";
import { statGroupFromItems } from "../entity/stat";
import * as Player from "../entity/playerAppearance";
import * as PlayerAppearance from "../entity/playerAppearance";

type StatGroup = Partial<Record<Stat.StatType, Stat.Stat[]>>;

export function allDataForPlayer(player: Player.PlayerId): StructuredQuery<{
    appearances: PlayerAppearance.PlayerAppearance[];
    stats: StatGroup;
}> {
    return {
        query: new QueryCommand({
            TableName: "inventory",
            KeyConditionExpression: "pk = :pk",
            ExpressionAttributeValues: {
                ":pk": `player#${player}`,
            },
        }),
        parser: (items: Record<string, any>[]) => {
            return {
                appearances: items
                    .filter((item) => item.entity_type === "player_appearance")
                    .map((appearanceStorage) => PlayerAppearance.fromStorage(appearanceStorage)),
                stats: statGroupFromItems(items),
            };
        },
    };
}
