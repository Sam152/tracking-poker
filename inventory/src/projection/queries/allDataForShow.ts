import { QueryCommand } from "@aws-sdk/lib-dynamodb";
import { StructuredQuery } from "./index";
import * as Stat from "../entity/stat";
import { statGroupFromItems } from "../entity/stat";
import * as Show from "../entity/show";
import * as PlayerAppearance from "../entity/playerAppearance";

type StatGroup = Partial<Record<Stat.StatType, Stat.Stat[]>>;

export function allDataForShow(showId: Show.ShowId): StructuredQuery<{
    show?: Show.Show;
    players: PlayerAppearance.PlayerAppearance[];
    stats: StatGroup;
}> {
    return {
        query: new QueryCommand({
            TableName: "inventory",
            IndexName: "gsi1",
            KeyConditionExpression: "gsi1pk = :gsi1pk",
            ExpressionAttributeValues: {
                ":gsi1pk": `slug#${showId}`,
            },
        }),
        parser: (items: Record<string, any>[]) => {
            const show = items.find((item) => item.entity_type === "show");
            return {
                show: show ? Show.fromStorage(show) : undefined,
                players: items
                    .filter((item) => item.entity_type === "player_appearance")
                    .map((appearanceStorage) => PlayerAppearance.fromStorage(appearanceStorage)),
                stats: statGroupFromItems(items),
            };
        },
    };
}
