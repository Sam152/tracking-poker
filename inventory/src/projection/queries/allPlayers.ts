import { QueryCommand } from "@aws-sdk/lib-dynamodb";
import { StructuredQuery } from "./index";
import * as Player from "../entity/player";

export function allPlayers(): StructuredQuery<{
    players: Player.Player[];
}> {
    return {
        query: new QueryCommand({
            TableName: "inventory",
            KeyConditionExpression: "pk = :pk AND begins_with(sk, :skBeginsWith)",
            ExpressionAttributeValues: {
                ":pk": `all_operators`,
                ":skBeginsWith": `player#`,
            },
        }),
        parser: (items: Record<string, any>[]) => {
            return {
                players: items
                    .filter((item) => item.entity_type === "player")
                    .map((playerStorage) => Player.fromStorage(playerStorage)),
            };
        },
    };
}
