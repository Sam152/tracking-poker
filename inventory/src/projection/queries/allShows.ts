import { QueryCommand } from "@aws-sdk/lib-dynamodb";
import { StructuredQuery } from "./index";
import * as Show from "../entity/show";

export function allShows(): StructuredQuery<Show.Show[]> {
    return {
        query: new QueryCommand({
            // Scan index backwards, to show most recent shows first.
            ScanIndexForward: false,
            TableName: "inventory",
            KeyConditionExpression: "pk = :pk AND begins_with(sk, :skBeginsWith)",
            ExpressionAttributeValues: {
                ":pk": `operator#hcl`,
                ":skBeginsWith": `show#`,
            },
        }),
        parser: (items: Record<string, any>[]) => {
            return items.map((storageShow) => Show.fromStorage(storageShow));
        },
    };
}
