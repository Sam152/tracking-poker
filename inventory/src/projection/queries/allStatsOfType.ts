import { QueryCommand } from "@aws-sdk/lib-dynamodb";
import { StructuredQuery } from "./index";
import * as Stat from "../entity/stat";

export function allStatsOfType(type: Stat.StatType): StructuredQuery<Stat.Stat[]> {
    return {
        query: new QueryCommand({
            TableName: "inventory",
            IndexName: "gsi2",
            KeyConditionExpression: "gsi2pk = :gsi2pk",
            ExpressionAttributeValues: {
                ":gsi2pk": `stat_type#${type}`,
            },
        }),
        parser: (items: Record<string, any>[]) => {
            return items.map((statStorage) => Stat.fromStorage(statStorage));
        },
    };
}
