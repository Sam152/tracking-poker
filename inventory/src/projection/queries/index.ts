import { QueryCommand } from "@aws-sdk/lib-dynamodb";

export type StructuredQuery<T> = {
    query: QueryCommand;
    parser: (items: Record<string, any>[]) => T;
};
