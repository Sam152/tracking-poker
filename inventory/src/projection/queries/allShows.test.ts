import { allShows } from "./allShows";
import { allShowsFixture } from "../__fixtures__/queries/allShowsFixture";

describe("allShows", () => {
    test("parser", () => {
        expect(allShows().parser(allShowsFixture)).toMatchSnapshot();
    });
});
