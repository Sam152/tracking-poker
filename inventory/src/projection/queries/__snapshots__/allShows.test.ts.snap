// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allShows parser 1`] = `
[
  {
    "date": "2023/06/15",
    "duration": 20165,
    "id": "dVs7ORBHsF0",
    "operator": "hcl",
    "show_name": "Sashimi, Kiran C2B, Israeli Ron & Mike X Play $10/20/40 - Commentary by Christian Soto",
  },
]
`;
